### PR TITLE
Minor Jenkins Pipeline Refactor

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR refactors the existing Jenkins pipelines so that it achieves the following:
* Main branches (feature-refactor, master, etc) will deploy correctly with the right parameterization
* Reduced code duplication by consolidating logic into a common pipeline groovy file
This PR also lowers the CPU Limit of Patroni Ephemeral pods from 1 core to 500 millicores.

[SHOWCASE-758](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-758)

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-128

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-128

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-128

File: openshift/commonPipeline.groovy [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+      if(DEBUG_OUTPUT.equalsIgnoreCase('true')) {
+        echo "DEBUG - Using project: ${openshift.project()}"
+      }
+
+      notifyStageStatus("Deploy - ${stageEnv}", 'PENDING')
+      createDeploymentStatus(projectEnv, 'PENDING', JOB_NAME, hostEnv, pathEnv)
+
+      echo "Checking for ConfigMaps and Secrets in project ${openshift.project()}..."
+      if(!(openshift.selector('cm', "getok-frontend-config").exists() &&
+      openshift.selector('cm', "getok-oidc-config").exists() &&
```
Comment (loneil) [inline openshift/commonPipeline.groovy]: Is this one still used in the new model?

Comment (jujaga) [reply openshift/commonPipeline.groovy]: It is not. However, until we merge to master, I want to leave these checks in place as deleting these config maps will make the master deploy fail to run. At this point, it's more for deployment safety insurance than anything else.

**Pull request merged**